modules/build/src/main/scala/scala/build/Artifacts.scala

```diff
@@ -83,6 +85,25 @@ object Artifacts {
     logger: Logger
   ): Either[BuildException, Artifacts] = either {
 
+    addClang.map { _ =>
+      val cache = FileCache()
+      val task = {
+
+        val artifact = Artifact(url = CLangInstaller.mambaBinaryUrl
+        // checksumUrls TODO: verify check sum of archive
+        ).withChanging(true)
+        cache.file(artifact).run.flatMap {
+          case Left(e: Exception) => Task.fail(e)
+          case Right(archive: File) =>
+            Task.delay {
+              Task.fromEither(CLangInstaller.install(os.Path(archive.toPath), logger))
+            }
```

Comment:
The `Task.delay(Task.fromEither(…))` looks fishy… You might need a `.flatMap`, like `Task.delay(Task.fromEither(…)).flatMap(identity)`